source/agora/node/Fee.d

```diff
@@ -229,6 +236,84 @@ public class FeeManager
         return calculateDataFee(data_size, this.TxPayloadFeeFactor);
     }
 
+    /***************************************************************************
+
+        Calculate the `Amount` of fees that should be paid to each Validator
+
+        Params:
+            tot_fee = Total amount of fees
+            tot_data_fee = Total amount of data fees
+            stakes = Staked UTXO of each Validator
+
+        Return:
+            `Amount` of fees that should be paid to each Validator
+
+    ***************************************************************************/
+
+    public Amount[] getValidatorFees (Amount tot_fee, Amount tot_data_fee,
+        UTXO[] stakes) nothrow @safe
+    {
+        // no stakes, no fees
+        if (stakes.length == 0)
```

Comment:
Technically I think this should be an assertion. There should always be active validators.